File: 2021/01/24/beomjo/algorithm/BOJ/java/1874.java

import java.io.*;
import java.util.Stack;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        Stack<Integer> stack = new Stack<>();
        int total = Integer.parseInt(br.readLine());
        int i = 1;
        while (total-- > 0) {
            int input = Integer.parseInt(br.readLine());
            while (i <= input) {
                stack.push(i);
                sb.append("+\n");
                i++;
            }

            if (stack.peek() == input) {
                stack.pop();
                sb.append("-\n");
            } else {
                System.out.print("NO");
                return;
            }
        }
        System.out.println(sb);
        br.close();
        return;
    }
}

File: 2021/01/24/beomjo/algorithm/BOJ/kotlin/1874.kt
import java.lang.StringBuilder
import java.util.*

fun main() {
    val stack = Stack<Int>()
    val sb = StringBuilder()
    var i = 1;
    repeat(readLine()?.toInt() ?: return) {
        val input = readLine()!!.toInt()
        while (i <= input) {
            stack.push(i)
            sb.append("+\n")
            i++
        }
        if (stack.peek() == input) {
            stack.pop()
            sb.append("-\n")
        } else {
            print("NO")
            return
        }
    }
    println(sb)
    return
}